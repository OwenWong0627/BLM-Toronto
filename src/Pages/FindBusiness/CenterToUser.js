import React from "react";

//Center to User's Location Button
function CenterToUser({ panTo, center }) {
   return (
      <button
         className="centerToUser"
         onClick={() => {
            panTo({
               lat: center.lat,
               lng: center.lng,
            });
         }}
      >
         <img src="/compass.svg" alt="compass" />
      </button>
   );
}

export default CenterToUser;