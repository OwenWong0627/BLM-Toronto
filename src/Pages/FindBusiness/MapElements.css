body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 18px;
}

/* for if you want to include a title */
.Map > h1 {
  position: absolute;
  user-select: none;
  top: 1rem;
  left: 1rem;
  color: #281414;
  z-index: 10;
  margin: 0;
  padding: 0;
}

.searchbar {
  position: absolute;
  top: 1rem;
  left: 50%;
  transform: translateX(-50%);
  width: 100%;
  max-width: 400px;
  z-index: 10;
  transition: 850ms;
}

.searchbar.shifted {
   left: 60%;
   transition: 350ms;
}

.searchbar input {
  padding: 0.5rem;
  font-size: 1.5rem;
  width: 100%;
}

.centerToUser {
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  background: none;
  border: none;
  z-index: 10;
}

.centerToUser img {
  width: 50px;
  cursor: pointer;
}

.home {
  position: absolute;
  bottom: 1.5rem;
  right: 3.5rem;
  background: none;
  border: none;
  z-index: 11;
  width: 65px;
  cursor: pointer;
}

.filterButton {
   width: 250px;
   height: 50px;
   border: none;
   outline: none;
   color: #fff;
   background: #00bfff;
   cursor: pointer;
   position: absolute;
   z-index: 0;
   border-radius: 10px;
   bottom: 0;
   left: 50%;
   -ms-transform: translate(-50%, -50%);
   transform: translate(-50%, -50%);
   font-size: 20px;
}

.filterButton:before {
    content: '';
    background: linear-gradient(45deg, #ff0000, #ff7300, #fffb00, #48ff00, #00ffd5, #002bff, #7a00ff, #ff00c8, #ff0000);
    position: absolute;
    top: -2px;
    left:-2px;
    background-size: 400%;
    z-index: -1;
    filter: blur(5px);
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    animation: glowing 20s linear infinite;
    opacity: 0;
    transition: opacity .3s ease-in-out;
    border-radius: 10px;
}

.filterButton:active {
    color: #000
}

.filterButton:active:after {
    background: transparent;
}

.filterButton:hover:before {
    opacity: 1;
}

.filterButton:after {
    z-index: -1;
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: #00bfff;
    left: 0;
    top: 0;
    border-radius: 10px;
}

@keyframes glowing {
    0% { background-position: 0 0; }
    50% { background-position: 400% 0; }
    100% { background-position: 0 0; }
}

@media screen and (max-width: 960px) {
   .searchbar.shifted {
      left: 65%;
   }
   .searchbar {
      max-width: 350px;
   }
}

@media screen and (max-width: 850px) {
   .searchbar.shifted{
      max-width: 300px;
      left: 70%;
   }
}

@media screen and (max-width: 728px) {
   .filterButton{
      width: 70vw;
   }
}

/* SIDEBAR */
.menu-button {
  display: flex;
  z-index: 9998;
  position: fixed;
  margin-left: 1rem;
  margin-top: 1rem;
  font-size: 2rem;
  background: none;
}

.menu-button.invisible {
  display: none;
}

.cross {
  color: #e4f6f8;
  margin-left: 19.5rem;
}

.side-menu {
  background-color: #fff;
  width: 350px;
  height: 100vh;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
  left: -100%;
  transition: 850ms;
  z-index: 9999;
}

.side-menu.active {
  left: 0;
  transition: 350ms;
}

.sidebar-header {
  background-color: #00bfff;
  width: 100%;
  height: 65px;
  display: flex;
}

.side-menu-items {
  width: 100%;
}

.side-menu-items h1 {
  text-align: center;
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
}

@media screen and (max-width: 728px) {
   .side-menu {
      width: 100vw;
   }
   .cross{
      margin-left: 95vw;
   }
}
