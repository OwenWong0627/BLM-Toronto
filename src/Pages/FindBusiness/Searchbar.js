import React from "react";

import usePlacesAutocomplete, {
   getGeocode,
   getLatLng,
} from "use-places-autocomplete";
import {
   Combobox,
   ComboboxInput,
   ComboboxPopover,
   ComboboxList,
   ComboboxOption,
} from "@reach/combobox";
import '@reach/combobox/styles.css';

//Search Bar
function Searchbar({ panTo, center }) {
   //this is a hook
   //ready just checks whether the google scripts are all loaded up
   //value means what the user is currently inputting
   //suggestions means the suggestions that were returned from the google API
   //setValue is a function to set the value
   //clearSuggestion clears all the suggestions
   const {
      ready,
      value,
      suggestions: { status, data },
      setValue,
      clearSuggestions,
   } = usePlacesAutocomplete({
      requestOptions: {
         location: { lat: () => center.lat, lng: () => center.lng },
         radius: 200000,
      },
   });

   return (
      <div className="searchbar">
         <Combobox
            onSelect={async (address) => {
               //the setvalue keeps the text u inputted in the search box without having to call the api
               setValue(address, false);
               clearSuggestions();

               try {
                  const results = await getGeocode({ address });
                  const { lat, lng } = await getLatLng(results[0]);
                  panTo({ lat, lng });
                  console.log(lat, lng);
               }
               catch {
                  console.log("ERROR");
               }
            }}
         >
            <ComboboxInput
               value={value}
               onChange={(event) => {
                  setValue(event.target.value);
               }}
               disabled={!ready}
               placeholder="Enter an address"
            />
            <ComboboxPopover>
               <ComboboxList>
                  {status === "OK" &&
                     data.map(({ place_id, description }) => (
                        <ComboboxOption key={place_id} value={description} />
                     ))}
               </ComboboxList>
            </ComboboxPopover>
         </Combobox>
      </div>
   );
}

export default Searchbar;