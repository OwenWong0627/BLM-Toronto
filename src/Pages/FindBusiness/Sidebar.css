.menu-button {
  display: flex;
  z-index: 9998;
  position: fixed;
  margin-left: 1rem;
  margin-top: 1rem;
  font-size: 2rem;
  background: none;
}

.menu-button.invisible {
  display: none;
}

.cross {
  color: #e4f6f8;
  margin-left: 19.5rem;
}

.side-menu {
  background-color: #fff;
  width: 350px;
  height: 100vh;
  display: flex;
  justify-content: center;
  position: fixed;
  top: 0;
  left: -100%;
  transition: 850ms;
  z-index: 9999;
}

.side-menu.active {
  left: 0;
  transition: 350ms;
}

.sidebar-header {
  background-color: #00bfff;
  width: 100%;
  height: 65px;
  display: flex;
}

.side-menu-items {
  width: 100%;
}

.side-menu-items h1 {
  text-align: center;
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
}

@media screen and (max-width: 728px) {
   .side-menu {
      width: 100vw;
   }
   .cross{
      margin-left: 95vw;
   }
}
