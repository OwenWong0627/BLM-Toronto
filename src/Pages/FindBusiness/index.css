body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 18px;
}

/* for if you want to include a title */
.Map > h1 {
  position: absolute;
  user-select: none;
  top: 1rem;
  left: 1rem;
  color: #281414;
  z-index: 10;
  margin: 0;
  padding: 0;
}

.search {
  position: absolute;
  top: 1rem;
  left: 50%;
  transform: translateX(-50%);
  width: 100%;
  max-width: 400px;
  z-index: 10;
}

.search input {
  padding: 0.5rem;
  font-size: 1.5rem;
  width: 100%;
}

.locate {
  position: absolute;
  top: 1rem;
  right: 1rem;
  background: none;
  border: none;
  z-index: 10;
}

.locate img {
  width: 30px;
  cursor: pointer;
}
.home {
  position: absolute;
  bottom: 1.5rem;
  right: 3.5rem;
  background: none;
  border: none;
  z-index: 11;
  width: 50px;
  cursor: pointer;
}
