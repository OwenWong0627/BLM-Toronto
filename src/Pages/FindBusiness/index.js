import React, { useRef, useCallback, useState, useEffect } from "react";
import { Link } from 'react-router-dom';
import {
   GoogleMap,
   useLoadScript,
   Marker,
   InfoWindow,
} from '@react-google-maps/api';

//The map styles is copied from snazzymaps.com
import mapStyles from './mapStyles';
import * as businessData from '../../businesses.json';
import './MapElements.css';
import './Sidebar.css';
import Checkbox from '../../Components/Checkbox';
import Searchbar from './Searchbar';
import CenterToUser from './CenterToUser';

const libraries = ["places"];

const mapContainerStyle = {
   height: "100vh",
   width: "100vw",
};
const options = {
   styles: mapStyles,
   disableDefaultUI: true,
   zoomControl: true,
};

const IPAPIURL = 'http://api.ipapi.com/api/check?access_key=dc3c71b823bbe9c2231225f4eea06429&fields=latitude,longitude';
/**
 * This is function that calls on a the free api from ipapi.com to get the latitude and longitude data
 * from the current user.
 * The fetched latitude and longitude data will be stored in the "center object" created above.
 * This api will avoid prompting the user to allow the "This app wants to know your location" function.
 * 
 * Because this operation takes a moment to execute but the google maps renders instantly,
 * I decided to add a delay to the rendering of the map to allow for this asynchronous function
 * to finish executing before rendering the map, or else the map will be centered in the ocean.
 * 
 * @async
 * @function getUserLocation
 * @param {string} URL - The URL to receive the user's location details from
 * @return {Promise<Object>} The latitude and longitude data from the URL.
 */
async function getUserLocation(URL) {
   let response = await fetch(URL);
   let data = await response.json();
   console.log(data);
   return {
      lat: data.latitude,
      lng: data.longitude
   }
}

function FindBusiness() {
   const { isLoaded, loadError } = useLoadScript({
      googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAPS_API_KEY,
      libraries,
   });

   const [sidebar, setSidebar] = useState(false);

   const showSidebar = () => (
      setSidebar(!sidebar)
   );

   const [center, setCenter] = useState({
      lat: 0,
      lng: 0
   });

   useEffect(() => {
      async function fetchLocationData() {
         const cntr = await getUserLocation(IPAPIURL);
         setCenter(cntr);
      }
      fetchLocationData();
   }, []);

   const [checkedItems, setCheckedItems] = useState({
      "Art": true,
      "Education": true,
      "Business": true,
      "Health": true,
      "RealEstate": true,
      "Shopping": true,
      "Community": true,
      "Hair": true,
      "Media": true,
      "Restaurant": true,
      "Finance": true,
      "Other": true,
   });

   const checkboxes = [
      { name: "Art", value: "Art, Artists & Art Galleries" },
      { name: "Education", value: "Education, Books & Black Authors" },
      { name: "Business", value: "Business, Services & Technology" },
      { name: "Health", value: "Doctors, Health & Fitness" },
      { name: "RealEstate", value: "Real Estate & Home Services" },
      { name: "Shopping", value: "Shopping" },
      { name: "Community", value: "Community & Faith Centers" },
      { name: "Hair", value: "Hair, Barbers & Beauty" },
      { name: "Media", value: "Media, Events & Entertainment" },
      { name: "Restaurant", value: "Restaurants, Bakeries & Grocery" },
      { name: "Finance", value: "Financial & Legal Services" },
      { name: "Other", value: "Travel, Auto & Other Services" }
   ];

   const handleChange = (event) => {
      console.log(event.target);
      console.log(event.target.checked);
      setCheckedItems({ ...checkedItems, [event.target.name]: event.target.checked });
   }

   const [selectAll, setSelectAll] = useState(true);
   const handleAllChecked = (event) => {
      let checked = event.target.checked;
      setSelectAll(checked);
      console.log(event.target);
      console.log(checked);
      Object.keys(checkedItems).forEach((key) => { checkedItems[key] = checked });
   }
   //This state will be just a base array with all the initial businesses, this state will be checked
   //everytime a user applies the filter
   const [allBaseMarkers, setAllBaseMarkers] = useState([]);

   const initializeMarkers = () => {
      const res = businessData.inPersonBusinesses.map((business) => {
         return {
            id: business.ID,
            name: business.NAME,
            type: business.TYPE,
            latitude: business.LATITUDE,
            longitude: business.LONGITUDE,
            address: business.ADDRESS,
            postalCode: business.POSTAL_CODE,
            city: business.CITY,
            website: business.WEBSITE,
            description: business.DESCRIPTION,
            contactInfo: business.CONTACT_INFO
         }
      })
      //TODO: RENAME DESCRIPTIONS to DESCRIPTION
      console.log(res);
      setAllBaseMarkers(res);
      setAllMarkers(res);
   }
   //This useEffect will just initialize the allMarkers and allBaseMarkers States
   useEffect(initializeMarkers, []);

   //This is the state that will actually be modified when filters are applied
   const [allMarkers, setAllMarkers] = useState([]);
   const hideMarkers = () => {
      let availableCategoryArray = []
      const availableCategoryObject = checkboxes.filter((option) => {
         if (checkedItems[option.name]) {
            return option;
         }
         return null;
      });
      Object.values(availableCategoryObject).forEach((value) => {
         availableCategoryArray.push(value.value);
         console.log(value);
      });
      console.log(availableCategoryObject);
      console.log(availableCategoryArray);
      console.log(allMarkers);
      const loadNewMarkerSet = allBaseMarkers.filter((business) => {
         if (availableCategoryArray.includes(business.type)) {
            return business;
         }
         return null;
      });
      console.log(loadNewMarkerSet);
      setAllMarkers(loadNewMarkerSet);
   }

   const [selectedBusiness, setSelectedBusiness] = useState(null);

   //This function listens to if user clicks the escape key, ifW pressed, the info window will close
   useEffect(() => {
      const listener = e => {
         if (e.key === "Escape") {
            setSelectedBusiness(null);
         }
      };
      window.addEventListener("keydown", listener);

      return () => {
         window.removeEventListener("keydown", listener);
      };
   }, []);

   const mapRef = useRef();
   const onMapLoad = useCallback((map) => {
      mapRef.current = map;
   }, []);

   const panTo = useCallback(({ lat, lng }) => {
      mapRef.current.panTo({ lat, lng });
      mapRef.current.setZoom(11);
   }, []);

   if (loadError) return "Error loading maps";
   if (!isLoaded || center.lat === 0) return "Loading...";

   return (
      <>
         <div className={sidebar ? 'menu-button invisible' : 'menu-button'} id='menu' onClick={showSidebar}>
            <i className={'fas fa-bars'} />
         </div>
         <div className={sidebar ? 'side-menu active' : 'side-menu'}>
            <ul className='side-menu-items'>
               <li className='sidebar-header'>
                  <div className='menu-button cross' onClick={showSidebar}>
                     <i className={'fas fa-times'} />
                  </div>
               </li>
               {/* The actual filter section */}
               <li>
                  <h1>Categories</h1>
               </li>
               <li><label><input type='checkbox' checked={selectAll} onChange={handleAllChecked} />Check/Uncheck All</label></li>
               {checkboxes.map(item => (
                  <Checkbox
                     key={item.name}
                     id={item.id}
                     name={item.name}
                     value={item.value}
                     checked={checkedItems[item.name]}
                     onChange={handleChange}
                  />
               ))}
            </ul>
            <button
               type="button"
               className="filterButton"
               onClick={() => { hideMarkers(); showSidebar() }}
            >
               Apply Filter
               </button>
         </div>
         <div className="Map">
            {/* <h1>BLM-Toronto <i className='fas fa-thumbs-up' /></h1> */}
            {/* Home Button */}
            <Link to="/">
               <img className='home' src='/HomeButton.svg' alt='Home-Button'></img>
            </Link>

            <CenterToUser panTo={panTo} center={center} />
            <Searchbar panTo={panTo} center={center} className={sidebar ? 'searchbar shifted' : 'searchbar'} />

            <GoogleMap
               mapContainerStyle={mapContainerStyle}
               zoom={9}
               center={center}
               options={options}
               onLoad={onMapLoad}
            >
               <Marker
                  animation={window.google.maps.Animation.DROP}
                  position={{
                     lat: center.lat,
                     lng: center.lng
                  }}
                  icon={{
                     url: `/HomeMarker.svg`,
                     scaledSize: new window.google.maps.Size(50, 50)
                  }}
                  onClick={() => panTo({
                     lat: center.lat,
                     lng: center.lng,
                  })}
               />
               {/* Maps Out All the Markers */}
               {allMarkers.map(business => (
                  <Marker
                     key={business.id}
                     category={business.type}
                     position={{
                        lat: parseFloat(business.latitude),
                        lng: parseFloat(business.longitude)
                     }}
                     onClick={() => {
                        setSelectedBusiness(business);
                        if (sidebar) {
                           showSidebar();
                        }
                     }}
                     icon={{
                        url: `/compass.svg`,
                        origin: new window.google.maps.Point(0, 0),
                        anchor: new window.google.maps.Point(12.5, 0),
                        scaledSize: new window.google.maps.Size(25, 25)
                     }}
                  />
               ))}
               {/* Info Window */}
               {selectedBusiness && (
                  <InfoWindow
                     onCloseClick={() => {
                        setSelectedBusiness(null);
                     }}
                     position={{
                        lat: parseFloat(selectedBusiness.latitude),
                        lng: parseFloat(selectedBusiness.longitude)
                     }}
                  >
                     <div>
                        <h2>{selectedBusiness.name}</h2>
                        <h3>{selectedBusiness.city}</h3>
                        <h3>{selectedBusiness.type}</h3>
                        <h4>{selectedBusiness.address}, {selectedBusiness.postalCode}</h4>
                        <p>{selectedBusiness.description}</p>
                        <a target="_blank" rel="noopener noreferrer" href={selectedBusiness.website}>{selectedBusiness.website}</a>
                     </div>
                  </InfoWindow>
               )}
            </GoogleMap>
         </div>
      </>
   );
}

export default FindBusiness;
//componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.