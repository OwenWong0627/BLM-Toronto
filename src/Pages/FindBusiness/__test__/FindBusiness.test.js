import * as ReactModule from 'react';
import React from 'react';
import FindBusiness from '..';
import * as GoogleMapsAPI from '@react-google-maps/api';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const baseMockCenter = { lat: 43.1, lng: -79.2 };
afterEach(cleanup);
describe('Renders FindBusiness Map page without crashing', () => {
   it('Shows a loading text', () => {
      const { getByText } = render(<Router><FindBusiness /></Router>);
      expect(getByText(/Loading.../i)).toBeInTheDocument();
   });
   it('Removes the loading text and renders the Googlemap', () => {
      GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      ReactModule.useState = jest.fn()
         .mockImplementationOnce(initialState => [false, () => { }])
         .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
         .mockImplementation(initialState => [initialState, () => { }]);
      const { getByTestId } = render(<Router><FindBusiness /></Router>);
      expect(getByTestId("map")).toBeInTheDocument();
   });
   it('Removes the loading text and renders the Home Button', () => {
      GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      ReactModule.useState = jest.fn()
         .mockImplementationOnce(initialState => [false, () => { }])
         .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
         .mockImplementation(initialState => [initialState, () => { }]);
      const { getByTestId } = render(<Router><FindBusiness /></Router>);
      expect(getByTestId("home-button")).toBeInTheDocument();
   });
   it('Removes the loading text and renders the sidebar with the apply filter button and checkbox list', () => {
      GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      ReactModule.useState = jest.fn()
         .mockImplementationOnce(initialState => [false, () => { }])
         .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
         .mockImplementation(initialState => [initialState, () => { }]);
      const { getByTestId } = render(<Router><FindBusiness /></Router>);
      expect(getByTestId("sidebar")).toBeInTheDocument();
      expect(getByTestId("checkbox-filter")).toBeInTheDocument();
      expect(getByTestId("filter-button")).toBeInTheDocument();
   });
   it('Removes the loading text and renders the centerToUser button', () => {
      GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      ReactModule.useState = jest.fn()
         .mockImplementationOnce(initialState => [false, () => { }])
         .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
         .mockImplementation(initialState => [initialState, () => { }]);
      const { getByTestId } = render(<Router><FindBusiness /></Router>);
      expect(getByTestId("centerToUser-button")).toBeInTheDocument();
   });
   it('Removes the loading text and renders the search bar', () => {
      GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
      ReactModule.useState = jest.fn()
         .mockImplementationOnce(initialState => [false, () => { }])
         .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
         .mockImplementation(initialState => [initialState, () => { }]);
      const { getByTestId } = render(<Router><FindBusiness /></Router>);
      expect(getByTestId("search-bar")).toBeInTheDocument();
   });
});

it('matches snapshot', () => {
   GoogleMapsAPI.default.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
   GoogleMapsAPI.useLoadScript = jest.fn().mockReturnValue({ isLoaded: true, loadError: null });
   ReactModule.useState = jest.fn()
      .mockImplementationOnce(initialState => [false, () => { }])
      .mockImplementationOnce(initialState => [baseMockCenter, () => { }])
      .mockImplementation(initialState => [initialState, () => { }]);
   const tree = renderer.create(<Router><FindBusiness /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});