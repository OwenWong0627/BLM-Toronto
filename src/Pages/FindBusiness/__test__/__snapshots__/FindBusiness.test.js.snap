// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="menu-button"
    id="menu"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </div>,
  <div
    className="side-menu"
    data-testid="sidebar"
  >
    <ul
      className="side-menu-items"
      data-testid="checkbox-filter"
    >
      <li
        className="sidebar-header"
      >
        <div
          className="menu-button cross"
          onClick={[Function]}
        >
          <i
            className="fas fa-times"
          />
        </div>
      </li>
      <li>
        <h1>
          Categories
        </h1>
      </li>
      <li>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <h2
            className="checkAll"
          >
            Check/Uncheck All
          </h2>
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Art, Artists & Art Galleries"
            onChange={[Function]}
            type="checkbox"
          />
          Art, Artists & Art Galleries
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Education, Books & Black Authors"
            onChange={[Function]}
            type="checkbox"
          />
          Education, Books & Black Authors
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Business, Services & Technology"
            onChange={[Function]}
            type="checkbox"
          />
          Business, Services & Technology
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Doctors, Health & Fitness"
            onChange={[Function]}
            type="checkbox"
          />
          Doctors, Health & Fitness
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Real Estate & Home Services"
            onChange={[Function]}
            type="checkbox"
          />
          Real Estate & Home Services
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Community & Faith Centers"
            onChange={[Function]}
            type="checkbox"
          />
          Community & Faith Centers
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Hair, Barbers & Beauty"
            onChange={[Function]}
            type="checkbox"
          />
          Hair, Barbers & Beauty
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Media, Events & Entertainment"
            onChange={[Function]}
            type="checkbox"
          />
          Media, Events & Entertainment
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Restaurants, Bakeries & Grocery"
            onChange={[Function]}
            type="checkbox"
          />
          Restaurants, Bakeries & Grocery
        </label>
      </li>
      <li>
        <label
          data-testid="checkbox-text"
        >
          <input
            checked={false}
            data-testid="checkbox"
            name="Travel, Auto & Other Services"
            onChange={[Function]}
            type="checkbox"
          />
          Travel, Auto & Other Services
        </label>
      </li>
    </ul>
    <button
      className="filterButton"
      data-testid="filter-button"
      onClick={[Function]}
      type="button"
    >
      Apply Filter
    </button>
  </div>,
  <div
    className="Map"
    data-testid="map"
  >
    <a
      data-testid="home-button"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Home-Button"
        className="home"
        src="/BLM-Toronto-Logo.svg"
        title="Go To Home Page"
      />
    </a>
    <button
      className="centerToUser"
      data-testid="centerToUser-button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Home Marker"
        src="/HomeMarker.svg"
      />
    </button>
    <div
      className="searchbar"
    >
      <div
        data-reach-combobox=""
      >
        <input
          aria-autocomplete="both"
          aria-controls="listbox--7"
          aria-expanded={false}
          aria-haspopup="listbox"
          data-reach-combobox-input=""
          data-testid="search-bar"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter an address"
          role="combobox"
          value=""
        />
        <span />
      </div>
    </div>
  </div>,
]
`;
